### PR TITLE
F#19 Main Menu

### PR DESCRIPTION
**Pull request by Armadillq**
hopefully this one isn't conflicting:)

### CODE REVIEW

File: Assets/Scripts/MainMenu/AudioController.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+            }
+        }
+    }
+    /// <summary>
+    /// Plays the desired sound
+    /// </summary>
+    /// <param name="name">name of the sound</param>
+    public void Play(string name)
+    {
+        Sounds s = Array.Find(sounds, sounds => sounds.name == name);
```
Comment (massieb) [inline Assets/Scripts/MainMenu/AudioController.cs]: Fix possible nullpointer when sound is not found

File: Assets/Scripts/MainMenu/AudioController.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
+﻿using UnityEngine;
+using System;
+using UnityEngine.Audio;
+
+public class AudioController : MonoBehaviour
```
Comment (massieb) [inline Assets/Scripts/MainMenu/AudioController.cs]: Add comments what this class is used for

File: Assets/Scripts/MainMenu/MainMenuButton.cs [hunk 3]
```diff
@@ -0,0 +1,278 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using UnityEngine.Audio;
+using UnityEngine.SceneManagement;
+
+public class MainMenuButton : MonoBehaviour
```
Comment (massieb) [inline Assets/Scripts/MainMenu/MainMenuButton.cs]: Is this class for only 1 button on the main menu? That is what the name is saying now.

File: Assets/Scripts/MainMenu/MainMenuButton.cs [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+{
+    public GameObject mainScreen, settingsScreen, creditsScreen, displayScreen, audioScreen, controlsScreen, helpScreen, loadGameScreen;
+    public Toggle player1Toggle, player2Toggle;
+    public Toggle sfxToggle, musicToggle, overallToggle;
+    public Slider sfxSlider, musicSlider, overallSlider;
+
+    #region MainScreen
+    public void OnStoryModePress()
+    {
+        SceneManager.LoadScene("TestSinglePlayerLevel");
```
Comment (massieb) [inline Assets/Scripts/MainMenu/MainMenuButton.cs]: Make this scene be changable by unity (serializable variable) instead of hard coded string

File: Assets/Scripts/MainMenu/MainMenuButton.cs [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+        settingsScreen.SetActive(false);
+        displayScreen.SetActive(false);
+        audioScreen.SetActive(false);
+        controlsScreen.SetActive(false);
+        helpScreen.SetActive(false);
+        loadGameScreen.SetActive(false);
+        creditsScreen.SetActive(false);
+        #endregion
+        #region Display
+        //Finds resolitions and offers them in the settings menu
```
Comment (massieb) [inline Assets/Scripts/MainMenu/MainMenuButton.cs]: typo: `resolutions`

File: Assets/Scripts/MainMenu/MainMenuButton.cs [hunk 6]
```diff
@@ -0,0 +1,278 @@
...
+    /// Sets the quality setting
+    /// </summary>
+    /// <param name="qualityIndex">List of qualities</param>
+    public void SetQuality(int qualityIndex)
+    {
+        QualitySettings.SetQualityLevel(qualityIndex);
+    }
+
+    /// <summary>
+    /// Sets the display to fullscreen and windowed
```
Comment (massieb) [inline Assets/Scripts/MainMenu/MainMenuButton.cs]: Toggles the display between ...

File: Assets/Scripts/MainMenu/MainMenuButton.cs (hunk 4 above)
Comment (Dejin1) [reply Assets/Scripts/MainMenu/MainMenuButton.cs]: I agree this needs to be easily changeable in the editor

File: Assets/Scripts/MainMenu/AudioController.cs [hunk 7]
```diff
@@ -2,19 +2,30 @@
...
+        audioSource = GetComponent<AudioSource>();
         //Sets a list of sounds
         foreach (Sounds s in sounds)
         {
-            s.source = gameObject.AddComponent<AudioSource>();
-            s.source.clip = s.clip;
+                s.source = gameObject.AddComponent<AudioSource>();
+                s.source.clip = s.clip;
 
+            if (s.source.clip.name == "Yami Switch 1")
```
Comment (massieb) [inline Assets/Scripts/MainMenu/AudioController.cs]: Try to avoid hardcoded strings. Extract them to be class constants (easier to notice and to refactor)

File: Assets/Scripts/MainMenu/MainMenuButtons.cs [hunk 8]
```diff
@@ -5,17 +5,21 @@
...
+/// <summary>
+/// Controls the functionality of the Main Menu
+/// </summary>
+public class MainMenuButtons : MonoBehaviour
 {
     public GameObject mainScreen, settingsScreen, creditsScreen, displayScreen, audioScreen, controlsScreen, helpScreen, loadGameScreen;
     public Toggle player1Toggle, player2Toggle;
     public Toggle sfxToggle, musicToggle, overallToggle;
     public Slider sfxSlider, musicSlider, overallSlider;
+    public int ScenePosition = 3;
```
Comment (massieb) [inline Assets/Scripts/MainMenu/MainMenuButtons.cs]: why is this changed from a string to an int? Is that how the loadscene works?

File: Assets/Scripts/MainMenu/MainMenuButtons.cs [hunk 9]
```diff
@@ -144,7 +151,7 @@ public void Start()
         creditsScreen.SetActive(false);
         #endregion
         #region Display
-        //Finds resolitions and offers them in the settings menu
+        //Finds resolotions and offers them in the settings menu
```
Comment (massieb) [inline Assets/Scripts/MainMenu/MainMenuButtons.cs]: almost :) `resolutions`

File: Assets/Scripts/MainMenu/MainMenuButtons.cs (hunk 8 above)
Comment (Armadillq) [reply Assets/Scripts/MainMenu/MainMenuButtons.cs]: The scenes are given numbers in order of their position in the build settings, I figured it would be easier to just enter the number in which the scene is in instead of typing out the name.

File: Assets/Scripts/MainMenu/MainMenuButtons.cs (hunk 9 above)
Comment (Armadillq) [reply Assets/Scripts/MainMenu/MainMenuButtons.cs]: Im getting there lmao, couple more tries

File: Assets/Scripts/MainMenu/MainMenuButtons.cs (hunk 8 above)
Comment (Dejin1) [reply Assets/Scripts/MainMenu/MainMenuButtons.cs]: Yeah I agree with Arma the load scene does work fine with integers

Comment (massieb) [reply Assets/Scripts/MainMenu/MainMenuButtons.cs]: Numbers dont say anything to me in code. Than i would move them to a file with all constant about scenes. Or maybe even an Enum? Maybe hiddenRaccoon can suggest a method here? This is only an improvement and can also be changed later. The commeny i made is solved
